File: C/bulles.c
#include "stdlib.h"
#include "stdio.h"

void tri_bulles(int* tableau, int taille) {
    int i, j, tmp;
    for (i = 0; i < taille - 1; i++) {
        for (j = 0; j < taille - i - 1; j++) {
            if (tableau[j] > tableau[j + 1]) {
                tmp = tableau[j];
                tableau[j] = tableau[j + 1];
                tableau[j + 1] = tmp;
            }
        }
    }
}

int main() {

    int tableau[] = {5, 6, 3, 2, 1};

    tri_bulles(tableau, 5);

    printf("%d %d %d %d %d\n", tableau[0], tableau[1], tableau[2], tableau[3], tableau[4]);

    return EXIT_SUCCESS;

}

File: Python/bulles.py
def tri_bulles(liste):
    for i in range(len(liste)):
        for j in range(len(liste)-1):
            if liste[j] > liste[j+1]:
                liste[j], liste[j+1] = liste[j+1], liste[j]
    return liste

liste = [1, 5, 3, 2, 4]

liste = tri_bulles(liste)

print(liste)